editor/src/components/canvas/ui-jsx-canvas.tsx

```diff
@@ -454,7 +467,10 @@ export const UiJsxCanvas = React.memo<UiJsxCanvasPropsWithErrorCallback>((props)
     )
 
     // IMPORTANT this assumes createExecutionScope ran and did a full walk of the transitive imports!!
-    if (shouldRerenderRef.current) {
+    const isRemounted =
+      props.invalidatedCanvasData.mountCountInvalidated ||
+      props.invalidatedCanvasData.domWalkerInvalidated
```

Comment:
if we only clear the cache on `invalidatedCanvasData.mountCountInvalidated`, isn't it a problem that we `unimport...` on `domWalkerInvalidated` too?

Reply:
so when the `proejctContents` are changed, the cache is automatically invalidated (no need to clear it)